### PR TITLE
Use working jest option

### PR DESCRIPTION
**Pull request by michaelsmitha**
Things to consider before you submit the PR:

* [x] Are tests passing locally?
* [x] Are the files formatted correctly?

## Description

Get Jest unit tests running on the SDK

image

## Related PRs or issues

[AB#64086](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/64086)

### CODE REVIEW

File: appcenter-analytics/package.json [hunk 1]
```diff
@@ -43,9 +43,7 @@
   },
   "jest": {
     "preset": "react-native",
-    "setupFilesAfterEnv": [
-      "./jest/setup.js"
-    ],
+    "setupTestFrameworkScriptFile": "./jest/setup.js",
```
Comment (dhei) [inline appcenter-analytics/package.json]: Looks like [setupTestFrameworkScriptFile is deprecated in favor of setupFilesAfterEnv](https://jestjs.io/docs/en/configuration#setupfilesafterenv-array).

Comment (michaelsmitha) [reply appcenter-analytics/package.json]: @dhei `setupFilesAfterEnv` doesn't work with our current setup. I know @jackycodes was looking at it a while ago, I don't think he ever got it working.

Comment (dhei) [reply appcenter-analytics/package.json]: Do we know when jest will deprecate `setupTestFrameworkScriptFile`?

Comment (michaelsmitha) [reply appcenter-analytics/package.json]: @dhei no, but it looks like jest@^24.x.x fixes the issue. I'll upgrade jest instead - that's better.

**Comment by michaelsmitha** [comment]
@dhei done. I had to bump the `devDependencies` versions of `react, `react-native`,  and `jest` - I just made them the same versions that are in `appcenter-analytics` though

**Pull request merged**